plugins/che-plugin-analytics/che-plugin-analytics-wsagent/src/main/java/com/redhat/che/plugin/analytics/wsagent/AnalyticsManager.java

```diff
@@ -0,0 +1,437 @@
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.hash.Hashing;
+import com.google.inject.Singleton;
+import com.google.inject.name.Named;
+import com.segment.analytics.Analytics;
+import com.segment.analytics.messages.TrackMessage;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.text.MessageFormat;
+import java.util.Collections;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import org.eclipse.che.api.core.rest.HttpJsonRequestFactory;
+import org.eclipse.che.api.workspace.shared.dto.WorkspaceDto;
+import org.slf4j.Logger;
+
+/**
+ * Send event to Segment.com and regularly ping Woopra. For now the events are provided by the
+ * {@link UrlToEventFilter}.
+ *
+ * @author David Festal
+ */
+@Singleton
+public class AnalyticsManager {
+  private static final Logger LOG = getLogger(AnalyticsManager.class);
+
+  private static final String pingRequestFormat =
+      "http://www.woopra.com/track/ping?host={0}&cookie={1}&timeout={2}";
+
+  private final Analytics analytics;
+
+  private final long noActivityTimeout = 60000 * 3;
+  private final long pingTimeout = 30 * 1000;
+  private final String workspaceId;
+  private final String workspaceName;
+
+  private String segmentWriteKey;
+  private String woopraDomain;
+
+  private ScheduledExecutorService checkActivityExecutor =
+      Executors.newSingleThreadScheduledExecutor(
+          new ThreadFactory() {
+            @Override
+            public Thread newThread(final Runnable r) {
+              return new Thread(
+                  new Runnable() {
+                    @Override
+                    public void run() {
+                      r.run();
+                    }
+                  },
+                  "Analytics Activity Checker");
+            }
+          });
+
+  private ScheduledExecutorService networkExecutor =
+      Executors.newSingleThreadScheduledExecutor(
+          new ThreadFactory() {
+            @Override
+            public Thread newThread(final Runnable r) {
+              return new Thread(
+                  new Runnable() {
+                    @Override
+                    public void run() {
+                      r.run();
+                    }
+                  },
+                  "Analytics Network Request Submitter");
+            }
+          });
+
+  private LoadingCache<String, EventDispatcher> dispatchers;
+
+  private String workspaceStartingUserId = null;
+
+  @Inject
+  public AnalyticsManager(
```

Comment:
wdym by a user ? a plugin developer for example ?

Reply:
No, I mean a user of Che on OSIO can send any activity to Woopra if I'm not mistaken.  Does it have any protection?

Reply:
It seems to me it would be much easier for a user to send wrong data to Segment / Woopra from the  web application console if the integration was done in javascript on the client-side (just call some javascript method in the browser console). But here eveything is done inside Java code running inside the wsagent tomcat process, and using Segment / woopra settings that are finally stored in an OpenShift config map (or secret).

Reply:
Yes, it should be even easier from the client side. So maybe it is not a problem. 
I was asking because I didn't see any injection of any key or so from config map. But after discussion of the JS-based solution, it doesn't seem a problem to me.

Reply:
The keys are retrieved by the wsagent from a wsmaster service endpoint. And the wsmaster gets them finally from a config map.